From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Laura Abbott <labbott@redhat.com>
Date: Thu, 2 Jan 2020 14:24:47 -0500
Subject: [PATCH] mptsas: pci-id table changes

Author: Tomas Henzl <thenzl@redhat.com>
Date:   Thu Jan 31 16:54:51 2019 -0500

    [message] mptsas: pci-id table changes

    Message-id: <20190131165452.7562-3-thenzl@redhat.com>
    Patchwork-id: 239765
    O-Subject: [RHEL8 e-stor 2/3] mptsas: pci-id table changes
    Bugzilla: 1666730
    RH-Acked-by: Jarod Wilson <jarod@redhat.com>
    RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
    RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>

    RHEL-only

    We want to provide this driver only for the virtualisation use case,
    so we remove all other entries from the pci-id table and add only the
    single entry needed for virtualisation.

    Signed-off-by: Herton R. Krzesinski <herton@redhat.com>

Upstream Status: RHEL only
Signed-off-by: Laura Abbott <labbott@redhat.com>
---
 drivers/message/fusion/mptsas.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/message/fusion/mptsas.c b/drivers/message/fusion/mptsas.c
index c17614cd6ed8..ca44a35ec411 100644
--- a/drivers/message/fusion/mptsas.c
+++ b/drivers/message/fusion/mptsas.c
@@ -5358,6 +5358,10 @@ static void mptsas_remove(struct pci_dev *pdev)
 }

 static struct pci_device_id mptsas_pci_table[] = {
+#ifdef CONFIG_RHEL_DIFFERENCES
+	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1068,
+		PCI_VENDOR_ID_VMWARE, PCI_ANY_ID },
+#else
 	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1064,
 		PCI_ANY_ID, PCI_ANY_ID },
 	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1068,
@@ -5370,6 +5374,7 @@ static struct pci_device_id mptsas_pci_table[] = {
 		PCI_ANY_ID, PCI_ANY_ID },
 	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1068_820XELP,
 		PCI_ANY_ID, PCI_ANY_ID },
+#endif
 	{0}	/* Terminating entry */
 };
 MODULE_DEVICE_TABLE(pci, mptsas_pci_table);
-- 
2.26.2

